### ISSUE TITLE
config:set --lock-config does not act on other scopes

### ISSUE THREAD

**Issue by kassner** [magento/magento2#19609]
### Preconditions

1. Magento 2.2.7;

### Steps to reproduce

1. `php bin/magento config:set --lock-config general/store_information/name "Test lock"`;
2. `php bin/magento app:config:import`;
3. `php bin/magento cache:clean`;

### Expected result

Admin user should not be able to change the configuration.

### Actual result

Admin user is not able to change the configuration for the default store, but is able to change it for any other scope (i.e.: store view).

`app/etc/config.php` generated:
```php
<?php
return [
    'modules' => [
        // ...
    ],
    'system' => [
        'default' => [
            'general' => [
                'store_information' => [
                    'name' => 'Test lock'
                ]
            ]
        ]
    ]
];

```

### Comments

Using `--scope` and `--scope-code` with the `config:set` command, I can lock for that particular store as well, but then I need to duplicate all the settings for every single website and store view in the system, and it will not effect stores created after the configuration is locked. There is no way to tell Magento "don't change this configuration in *any* scope".

**Comment by magento-engcom-team** [comment]
Hi @kassner. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@kassner do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97006`, `MAGETWO-97007` were created

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @kassner. Thank you for your report. 
The issue has been fixed in magento/magento2#20322 by @maheshWebkul721 in 2.2-develop branch
Related commit(s):
 - [c72cbfc878925fb122b14df7c7aedae7f2cb5728](https://github.com/magento/magento2/commit/c72cbfc878925fb122b14df7c7aedae7f2cb5728)
 - [816be4b43af11f49aa345f4f92eb31e81a559b5a](https://github.com/magento/magento2/commit/816be4b43af11f49aa345f4f92eb31e81a559b5a)

The fix will be available with the upcoming 2.2.8 release.
